tests/karate/src/test/java/be/utils/Server.java

```diff
@@ -32,9 +38,33 @@ private static ProcessBuilder build(String[] cmd, String dir, String logPath) {
   }
 
   @Override
-  public void stop() {
-    process.children().forEach(ProcessHandle::destroyForcibly);
+  public void stop() throws RuntimeException {
+    process.descendants().forEach(ProcessHandle::destroy);
+    process.destroy();
+    List<CompletableFuture<?>> allProcesses = process.descendants().map(ProcessHandle::onExit).collect(Collectors.toList());
+    allProcesses.add(process.onExit());
+    CompletableFuture<Void> onExit = CompletableFuture.allOf(allProcesses.toArray(new CompletableFuture[]{}));
```

Comment:
this thing looks a bit cryptic to me. Can we improve the readability ?

```suggestion
    List<CompletableFuture<?>> allProcesses = process
      .descendants()
      .map(ProcessHandle::onExit)
      .collect(Collectors.toList());

    allProcesses.add(process.onExit());

    CompletableFuture<Void> onExit = CompletableFuture
      .allOf(allProcesses.toArray(new CompletableFuture[]{}));
```

Reply:
Sure